src/tespy/networks/networks.py

```diff
@@ -2511,6 +2511,134 @@ def postprocessing(self):
+        - Calculate exergy destruction and exergetic efficiency of components.
+          If there is a componenten in the network for which there is no
+          exergy balance implemented in TESPy, 0 is assigned to E_D, E_F and
+          E_P of the componenten. The exergetic efficiency epsilon gets the
+          value -111.
+        - Sum up exergy destruction of components with bottom up approach.
+        - The network product exergy equals the net power output.
+        - Assign network fuel exergy and exergy los from parameters given
+          to method.
+        - Calculate network exergy destruction and epsilon.
+        - Calculate exergy destruction ratios for components.
+
+        Parameters
+        ----------
+        pamb : float
+            Ambient pressure in Pa.
+        Tamb : float
+            Ambient temperature in K.
+        bus  : tespy.connections.bus
+            Energy flows in network. Used to calculate product exergy
+            or fuel exergy of turbines and pumps.
+        E_F  : list
+            List containing components which represent fuel exergy input
+            in network.
+        E_L  : list
+            List containing connections which represent exergy loss streams.
+
+        Note
+        ----
+            .. math::
+
+                E_{\text{D},comp} = E_{\text{F},comp} - E_{\text{P},comp}\\
+                \varepsilon_{\text{comp}} =
+                \frac{E_{\text{P},comp}}{E_{\text{F},comp}}
+
+                E_{\text{D,tot,bottom up}} = \sum_{comp=0}^N E_{\text{D},comp}
+
+                E_{\text{P}} = P_{\text{net}}\\
+                E_{\text{F}} = \sum_{comp=0}^N E_{\text{F},comp}\\
+                E_{\text{L}} = \sum_{conn=0}^N E_{\text{PH},conn}\\
+                E_{\text{D}} = E_{\text{F}} - E_{\text{P}} - E_{\text{L}}\\
+                \varepsilon = \frac{E_{\text{P}}}{E_{\text{F}}}
+
+                y_{\text{D},comp} =
+                \frac{\dot{E}_{\text{D},comp}}{\dot{E}_{\text{F}}}\\
+                y^*_{\text{D},comp} =
+                \frac{\dot{E}_{\text{D},comp}}{\dot{E}_{\text{D}}}
+        """
+        if E_F is None:
+            msg = ('Missing fuel exergy E_F of network.')
+            logging.warning(msg)
+
+        # physical exergy of connections
+        for conn in self.conns.index:
+            conn.get_physical_exergy(pamb, Tamb)
+
+        # exergy balance of components
+        for cp in self.comps.index:
+            if (isinstance(cp, cycle_closer) or isinstance(cp, sink) or
+                    isinstance(cp, source) or isinstance(cp, splitter)):
+                continue
+            cp.exergy_balance(Tamb, bus)
+
+            # calculate E_D for components
+            try:
+                if isinstance(cp.E_P, str):
+                    cp.E_D = cp.E_F
+                else:
+                    cp.E_D = cp.E_F - cp.E_P
+            except AttributeError:
+                cp.E_D = 0
+                cp.E_F = 0
+                cp.E_P = 0
```

Comment:
I would like to remove the try except. The exergy_balance base method can assign these values for components that do not yet have the exergy balance implemented. As it is called prior to this line, there should not be an issue.

Reply:
Alright, I will think of a solution there.